### PR TITLE
[NextPluggableDevice] expose necessary c headers

### PR DESCRIPTION
**Pull request by jzhoulon**
This PR is exposing those necessary c header files to use NextPluggableDevice C APi

### CODE REVIEW

File: tensorflow/tools/pip_package/BUILD
Comment (jyingl3) [inline tensorflow/tools/pip_package/BUILD:46]: Just to double check, is pjrt_c_api_tpu_hdrs required?

**Review by jyingl3 — COMMENTED** [review]
There may be size limit. Will pull in other reviewer if there is size issue after the size limit check triggered when merged.

**Comment by jyingl3** [comment]
Thanks Zhoulong!

@chuanhaozhuge FYI

File: tensorflow/tools/pip_package/BUILD
Comment (jzhoulon) [reply tensorflow/tools/pip_package/BUILD:46]: @jyingl3 Thanks for the review, pjrt_c_api_tpu.h contains the declaration of GetPjrtApi, which is entry point of plugin loaded by TF
```
 #include "tensorflow/compiler/xla/pjrt/c/pjrt_c_api.h"

 #ifdef __cplusplus
 extern "C" {
 #endif

 // TODO(b/261916900): this will cause symbol collision when we wanna statically
 // link in multiple plugins. We may need to make the function name configurable,
 // or we could add external-only wrapper shims
 // Does not pass ownership of returned PJRT_Api* to caller.
 const PJRT_Api* GetPjrtApi();

 #ifdef __cplusplus
 }
 #endif
```

Comment (jyingl3) [reply tensorflow/tools/pip_package/BUILD:46]: Thanks Zhoulong. We probably should move the declaration of GetPjrtApi or rename this file then. Meanwhile, I do not want to block this change so probably we can go ahead with the it and refactor later. @chuanhaozhuge Does it look good to you?

Comment (jyingl3) [reply tensorflow/tools/pip_package/BUILD:46]: Discussed with @stellaraccident. We think the plugin will likely have its own declaration of GetPjrtApi to add attributes to control/export/etc. And we will add the function typedef with documentation on the name (GetPjrtApi) to pjrt_c_api.h as a guidance.

In this case, could we not expose "//tensorflow/compiler/xla/pjrt/c:pjrt_c_api_tpu_hdrs" here, and let the plugin to has its own declaration of GetPjrtApi?

Comment (jzhoulon) [reply tensorflow/tools/pip_package/BUILD:46]: @jyingl3  thanks for the suggestion, I removed the pjrt_c_api_tpu_hdrs, we are ok to declare the GetPjRtApi in plugin side.
Thanks

Comment (jyingl3) [reply tensorflow/tools/pip_package/BUILD]: Thanks Zhoulong! I will approve it. May pull in other reviewer if it increases the size too much.

**Review by jyingl3 — APPROVED** [review]
@chuanhaozhuge @qqfish @penpornk  FYI for this change.

**Pull request merged**
